# Необходимо запустить на исполнение расчетную задачу task в пуле процессов. Известно, что функция task принимает три позиционных аргумента task(a, b, c).
# Аргументы необходимо взять из источника - генератора data_gen. Известно, что функция - генератор data_gen возвращает кортеж из трех элементов.
# Также известно, что функция task может вызывать различные исключения из группы Exception во время своей работы.

# Допишите код так, чтобы вывести переменную result в виде списка результатов. Порядок в списке должен соответствовать порядку генерации аргументов
# функцией data_gen. Если функция task во время своей работы возбуждает исключение, вместо результата функции в списке results должна быть строка с
# типом исключения и сообщением.


from multiprocessing.pool import Pool


def wrapper(*args):
    try:
        return task(*args)
    except Exception as err:
        return err

if __name__ == '__main__':
    with Pool() as pool:
        result = pool.starmap(wrapper, data_gen())
        print(result)